### ISSUE TITLE
Improve comments in code

### ISSUE THREAD

**Issue by jonblack** [cmbi/kmad#31]
There are a lot of comments but they give much useful information. For example:

```
  ///
  /// Holds a single residue codon, and a list of its features
  ///
  struct Residue {
    Residue(std::string codon, std::vector<std::string> features) 
      : codon(codon),
        features(features) {}
    Residue(std::string codon) : codon(codon) {}
    Residue(){};

    std::string codon;
    std::vector<std::string> features;
  };
```

I recall you explaining the each "residue" in your fasta format has 3 values per residue. Is that what you mean by "codon" here? An example would help.

Another example:

```
  ///
  /// takes an alignment and residues to remove; returns an alignment with
  /// removed residues
  ///
  std::vector<fasta::SequenceList> move_residues(
      const std::vector<fasta::SequenceList>& alignment,
      const std::vector<MoveData>& move_data);
```

The function is called `move_residues` but the comment mentions *removing*. What I totally miss is **why** residues need moving or removing.

**Comment by joannalange** [comment]
Will fix it (second one example is just me being stupid, that's a typo, function should be called remove_residues)

You mean comments should be improved in the whole code not just these two things, right?

**Comment by jonblack** [comment]
Yup, everywhere. Concentrate on public functions first, in the headers. That helps to get an overview. Probably better to work on other issues in #32 first. And for that, we should probably sit down together for some pair-programming.
